test/functional/url-expander/test-expander.js

```diff
@@ -97,29 +97,53 @@ describes.realWin('Expander', {
       ABC: () => 'three',
       ABCD: () => 'four',
     };
-    beforeEach(() => {
-      env.win.document.head.appendChild(
-          createElementWithAttributes(env.win.document, 'meta', {
-            name: 'amp-allowed-url-macros',
-            content: 'ABC,ABCD,CANONICAL',
-          }));
-
-      variableSource = new GlobalVariableSource(env.ampdoc);
-      expander = new Expander(variableSource);
-    });
 
-    it('should not replace unwhitelisted RANDOM', () => {
-      const url = 'http://www.google.com/?test=RANDOM';
-      const expected = 'http://www.google.com/?test=RANDOM';
-      return expect(expander.expand(url, mockBindings))
-          .to.eventually.equal(expected);
+    describe('Non-empty whitelist', () => {
```

Comment:
nitpick: can you flatten the extra nested describe?

for example,
```
describe('Whitelist of variables', () => {
    ...
    beforeEach(() => {
        variableSource = new GlobalVariableSource(env.ampdoc);
        expander = new Expander(variableSource);
    });
    function addAllowedUrlMacroMeta(metaContent) {
        ...
    }
    it('should not replace unwhitelisted RANDOM', () => {
         addAllowedUrlMacroMeta('ABC,ABCD,CANONICAL');
         ...
    });
});
```

Reply:
Done. I had to create the expander after the whitelist meta tag is inserted into the DOM because the whitelist is read by the expander's constructor.